src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/RichTextEditorTextArea.kt

```diff
@@ -0,0 +1,59 @@
+package com.atlassian.performance.tools.jiraactions.api.page
+
+import com.atlassian.performance.tools.jiraactions.Patience
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.openqa.selenium.By
+import org.openqa.selenium.JavascriptExecutor
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.ExpectedConditions
+import org.openqa.selenium.support.ui.ExpectedConditions.*
+import java.time.Duration
+
+/**
+ * Helper class for Rich Text Editor enabled Textarea
+ *
+ */
+class RichTextEditorTextArea(
+    private val driver: WebDriver
+) {
+
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+
+    private fun clearAndTypeIfPresent(id :String, text: String) {
+        if (driver.isElementPresent(By.id(id))) {
+            val webElement = driver.findElement(By.id(id))
+            driver.wait(timeout = Duration.ofSeconds(2), condition = elementToBeClickable(webElement))
+            webElement.clear()
+            webElement.sendKeys(text)
+        }
+    }
+
+    fun clearAndTypeIfPresentRte(id : String, text: String) {
+        //check richeditor-cover exists in class attribute
+        if (driver.isElementPresent(By.id(id))) {
+            val webElement = driver.findElement(By.id(id))
+            val attrClass = webElement.getAttribute("class")
+            if(attrClass != null && attrClass.contains(other = "richeditor-cover", ignoreCase = true)){
+                logger.debug("richeditor detected, will try Rich Text Editer...")
+                val iframeXpath = "//div[textarea[@id='$id']]//iframe"
```

Comment:
Is `//div[textarea[@id='$id']]` the same as `webElement`?
Would `webElement.findElement(By.tagName("iframe")` work?

Reply:
find by iframe will return more than one, e.g. Environment, Description, Comment etc.
We need to find the one associated with the target textarea

Reply:
That's why I suggested to call it on the `webElement` instead of `driver`

Reply:
sorry, I did make it clear.
The iFrame is not a sub element of Textarea field, the iFrame and the associated Textarea are sharing a same Div as parent element. We may could obtain the parent Div element, but I think there is not much difference compare to just call driver.find.